spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1333,12 +1337,14 @@ protected void pollAndInvoke() {
 				return;
 			}
 			debugRecords(records);
-			resumeConsumerIfNeccessary();
-			if (!this.consumerPaused) {
-				resumePartitionsIfNecessary();
-			}
 
 			invokeIfHaveRecords(records);
+			if (this.pendingRecordsAfterError == null) {
```

Comment:
Just a thought. Since we're not committed to listening to `pause`/ `resume` calls while in this logic, I wonder if maybe we should also ignore it for partition pausing / resuming, and instead handle back offs relying on the `KafkaBackOffException` itself. The exception contains information such as the `TopicPartition` that was backed off. Of course, this or any other solution can be part of a different PR.

Thanks.